src/cdk/drag-drop/drag-events.ts

```diff
@@ -52,13 +52,13 @@ export interface CdkDragExit<T = any, I = T> {
 
 
 /** Event emitted when the user drops a draggable item inside a drop container. */
-export interface CdkDragDrop<T, O = T> {
+export interface CdkDragDrop<T, O = T, I = T> {
```

Comment:
This appears to be a breaking API change that breaks 10 or so projects in google3. The default generic for `CdkDrag` is `any`, so if we want this to not be breaking we should make the default for `I` `any` as well.

Reply:
Oh, I see Jeremy already commented about this. I don't think people had incorrect typings though, they just didn't have any good way to specify the item type. After this change they can, but the default needs to be something that won't break people (i.e. `any`)

Reply:
The default here doesn't seem right to me either. 

Today, I may have a `CdkDragDrop<string[]>` which correctly types the container as `CdkDropList<string[]>` and the `item` is just `CdkDrag` where the type of its `data` is `any` (ideally its `string`). 

With this change, now by default, the `CdkDrag` types the data as `string[]` which isn't right. I don't think there would ever be a case where `I = T` is right (T should be the array type, I is the item itself)

It seems for now that Miles' suggestion would be best - rather than use `I = T`, we should have `I = any`.